packages/vscode-extension/src/connectionService.ts

```diff
@@ -462,8 +487,11 @@ function attachSchemaPollerConnectionFailedEventListeners(): void {
  * Attaches an event listener to handle connection errors.
  * This event is only handled once.
  */
-function attachSchemaPollerConnectionErrorEventListener(): void {
+function attachSchemaPollerConnectionEventListeners(): void {
   const schemaPoller = getSchemaPoller();
+  schemaPoller.events.on('schemaFetched', () => {
+    databaseInformationTreeDataProvider.refresh();
+  });
```

Comment:
Does this really guarantee the data inside the language server and the one shown in VSCode is in sync? Am I right understanding we have two schema pollers and one could not be in sync with the other?

Reply:
no, this only really guarantees that the vscode UI is in sync with the data that's been polled in the instance of the schema poller that it itself uses

you're right that we have 2 schema pollers, and the vscode extension acting as a kind of message broker for when the language server should make changes to its own schema poller instance (i.e. we've disconnected, or switched databases, etc)

I had this idea that we could build in a "runtime strategy" to the language server.. if it's being consumed by the vscode extension, it could share a schema poller instance, and hook in to the same events, as the extension. If it's running in a standalone way, it could create its own schema poller instance and operate in the same way that it currently does